### ISSUE TITLE
custom field slug with '/' might break UI

### ISSUE THREAD

**Issue by sklemer1** [nautobot/nautobot#1962]
### Environment
* Python version: 3.7.3
* Nautobot version: 1.3.2

### Steps to Reproduce
1. Create a new custom field named `Devices / Virtual Machines`
2. Go to https://mynautobot/extras/custom-fields/
3. You get
```
<class 'django.urls.exceptions.NoReverseMatch'>

Reverse for 'customfield' with arguments '('Devices / Virtual Machines',)' not found. 1 pattern(s) tried: ['extras/custom\\-fields/(?P<name>[^/]+)/\\Z']

Python version: 3.7.3
Nautobot version: 1.3.2
```

### Expected Behavior

Impossibility to add such a slug and/or correct URL reversal.

### Observed Behavior

Python Exception and need to remove the broken cf via nbshell.

## TODO 

- Add a slug field and use it in URL patterns

**Comment by glennmatthews** [comment]
Thanks for the report! #824 is the tracking issue for changing the CustomField model to actually make this field a proper slug field, which would fix this issue and several related symptoms.

**Comment by bryanculver** [comment]
As a bug fix, we will add a slug field and use it in URL patterns.

**Comment by glennmatthews** [comment]
Fixing this the "right" way will require adding a versioned REST API serializer, I think - so moving this to 1.4 milestone.

**Comment by glennmatthews** [comment]
Implemented for 1.4 by #2136.
